File: TCDNew-master/src/ui.android/lib/actions/data/purchase/PurchaseTaskLineUpdatePost.dart
import 'dart:async';

import 'package:flutter/widgets.dart';
import 'package:tiq/actions/ui/utility/ShowModalMessage.dart';
import 'package:tiq/common/Messages.dart';
import 'package:tiq/models/dtos/PurchaseTaskLineUpdateDto.dart';
import 'package:tiq/models/dtos/PurchaseTaskLineUpdatePostReq.dart';
import 'package:tiq/models/exceptions/RpcException.dart';
import 'package:tiq/models/exceptions/UnauthorizedAccessException.dart';
import 'package:tiq/services/GStateProvider.dart';
import 'package:tiq/services/RpcService.dart';

class PurchaseTaskLineUpdatePost {
  Future<bool> run(BuildContext context, String purchaseTaskId, String productBarcode, String currentPalletCode, PurchaseTaskLineUpdateDto purchaseTaskLineUpdate) async {
    bool result;

    try {
      if (!GStateProvider.instance.stateAuth.isAuthorised) throw UnauthorizedAccessException();

      result = await RpcService.instance.purchaseTaskLineUpdatePost(PurchaseTaskLineUpdatePostReq(
          purchaseTaskId, productBarcode, currentPalletCode, purchaseTaskLineUpdate), GStateProvider.instance.stateAuth.authToken);
    }
    on RpcException catch(exception) {
      await ShowModalMessage().run(context, Messages.ErrorServer, exception.errors.map((e) => e.errorText).join('\r\n'));
    }
    on TimeoutException {
      await ShowModalMessage().run(context, Messages.ErrorServer, Messages.ErrorTimeout);
    }
    on NoSuchMethodError {
      await ShowModalMessage().run(context, Messages.ErrorUnknown, Messages.ErrorDataStructure);
    }
    catch (exception) {
      await ShowModalMessage().run(context, Messages.ErrorUnknown, Messages.ErrorPurchaseTaskLineUpdatePost);
    }

    return result;
  }
}

File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Actions/Data/Purchase/PurchaseTaskLineUpdatePost.cs
using System;
using System.Linq;
using System.Net;
using Cen.Wms.Client.Actions.UI.Utility;
using Cen.Wms.Client.Common;
using Cen.Wms.Client.Models.Dtos;
using Cen.Wms.Client.Models.Exceptions;
using Cen.Wms.Client.Services;
using NLog;

namespace Cen.Wms.Client.Actions.Data.Purchase
{
    class PurchaseTaskLineUpdatePost
    {
        public static bool Run(string purchaseTaskId, string productBarcode, string currentPalletCode, PurchaseTaskLineUpdateDto purchaseTaskLineUpdate)
        {
            bool result = false;

            try
            {
                if (!GStateProvider.Instance.StateAuth.IsAuthorised)
                    throw new UnauthorizedAccessException();

                result = RpcService.Instance.PurchaseTaskLineUpdatePost(
                    new PurchaseTaskLineUpdatePostReq
                    {
                        PurchaseTaskId = purchaseTaskId,
                        ProductBarcode = productBarcode,
                        CurrentPalletCode = currentPalletCode,
                        PurchaseTaskLineUpdate = purchaseTaskLineUpdate
                    },
                    GStateProvider.Instance.StateAuth.AuthToken
                );
            }
            catch (RpcException exception)
            {
                ShowModalMessage.Run(Messages.ErrorServer, string.Join(Environment.NewLine, exception.Errors.Select(e => e.ErrorText).ToArray()));
            }
            catch (WebException exception)
            {
                var logger = LogManager.GetLogger(Messages.LoggerNetName);
                logger.Error(exception);

                ShowModalMessage.Run(Messages.ErrorTransport, Messages.ErrorPurchaseTaskLineUpdatePost);
            }
            catch (Exception exception)
            {
                var logger = LogManager.GetLogger(Messages.LoggerLocalName);
                logger.Error(exception);

                ShowModalMessage.Run(Messages.ErrorUnknown, Messages.ErrorPurchaseTaskLineUpdatePost);
            }

            return result;
        }
    }
}
